main.tf

```diff
@@ -178,7 +178,7 @@ resource "aws_elasticache_replication_group" "default" {
   engine_version             = var.engine_version
   at_rest_encryption_enabled = var.at_rest_encryption_enabled
   transit_encryption_enabled = var.transit_encryption_enabled
-  transit_encryption_mode    = var.transit_encryption_mode
+  transit_encryption_mode    = var.transit_encryption_enabled ? var.transit_encryption_mode : null
```

Comment:
I think with `var.transit_encryption_mode` being `null` by default the condition isn't needed. It effectively duplicates logic in provider/AWS API.

I prefer to let user set unsupported parameter combination and receive proper error message, rather than masking it behind condition.

Reply:
Hey, sorry about that 😓 you are right I forgot about it when pushed the last change.

Should be fixed now. Can you test if it works now?